### ISSUE TITLE
False-positive unsupported-delete-operation

### ISSUE THREAD

**Issue by chlndh** [pylint-dev/pylint#8068]
Here a scenario were a false-positive warning is generated with Pylint 2.13.7

test.py
```python
class A:
    def __init__(self, b=False): 
        self._m = [] if b else None

    def reset(self):
        if self._m:
            del self._m[:]
```

output:
```
|test.py|7|16|error|E1138|unsupported-delete-operation|A.reset|test|'self._m' does not support item deletion|
```

**Comment by Pierre-Sassoulas** [comment]
I can reproduce in pylint 2.16.0 with the latest astroid supporting is None constraint.

Though if the code was:
```python
class A:

    def __init__(self, b=False): 
        self._m = [] if b else None

    def reset(self):
        if self._m is not None:
            del self._m[:]
```
Then the error would not be raised.
